es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java

```diff
@@ -158,19 +148,42 @@ public ElasticSearchDAOV5(Client elasticSearchClient, Configuration config, Obje
                 new LinkedBlockingQueue<>());
     }
 
-    private void updateIndexName(Configuration config) {
-        this.logIndexPrefix = config.getProperty("workflow.elasticsearch.tasklog.index.name", "task_log");
-        this.logIndexName = this.logIndexPrefix + "_" + SIMPLE_DATE_FORMAT.format(new Date());
+    @Override
+    public void setup() throws Exception {
+
+        elasticSearchClient.admin().cluster().prepareHealth().setWaitForGreenStatus().execute().get();
 
         try {
-            elasticSearchClient.admin().indices().prepareGetIndex().addIndices(logIndexName).execute().actionGet();
+
+            initIndex();
+            updateIndexName();
+            Executors.newScheduledThreadPool(1).scheduleAtFixedRate(() -> updateIndexName(), 0, 1, TimeUnit.HOURS);
```

Comment:
Can you please help me understand if there is any specific advantage of using newScheduledThreadPool(1) over https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html#newSingleThreadExecutor() . 

Also, please consider moving delay and interval values to constants. Again, this would result in changing the configurations easily and having smaller diffs going forward.

Reply:
Will move the delay and interval values to constants.
About your question, not sure If I am the right person to answer it. This code is already on branch 2.0, this PR is just moving code around in order to support a setup for IndexDao.
I can also change it, dont see why not use `newSingleThreadExecutor`, but I need to read over documentation and might delay my code update.